### PR TITLE
Issue1070 durack1 update pyspharm

### PR DESCRIPTION
**Pull request by durack1**
@doutriaux1 this updates #1074 to be rebased to current `master` (for additional comments take a peek at #1074) - `spherepack` will need to be added as a new package and `pyspharm_deps.cmake` updated so it's included as a dependency

### CODE REVIEW

**Comment by durack1** [comment]
@doutriaux1 just to get this one done and dusted, `spherepack3.2.tar` is now uploaded and ready to be sourced from `$LLNL_URL`

Do you need any other help with this?

`pyspharm 1.0.8` is still the current version

File: CMake/cdat_modules/pyspharm_pkg.cmake [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
+set(PYSPHARM_PATCH 8)
 set(PYSPHARM_VERSION ${PYSPHARM_MAJOR}.${PYSPHARM_MINOR}.${PYSPHARM_PATCH})
 set(PYSPHARM_URL ${LLNL_URL})
 set(PYSPHARM_GZ pyspharm-${PYSPHARM_VERSION}.tar.gz)
-set(PYSPHARM_MD5 e19897ec64d9e861ef6b93bf649a433c)
+set(PYSPHARM_MD5 7b3a33dd3cbeaa4b8bf67ed5bd210931)
 set(PYSPHARM_SOURCE ${PYSPHARM_URL}/${PYSPHARM_GZ})
 
-add_cdat_package_dependent(pyspharm "" "" OFF "CDAT_BUILD_LEAN" OFF)
+if (CDAT_BUILD_ALL)
```
Comment (doutriaux1) [inline CMake/cdat_modules/pyspharm_pkg.cmake]: @sankhesh we should do that for rpy2, I noticed it doesn't build if BUILD_MODE=ALL

**Comment by durack1** [comment]
@doutriaux1 just pinging you on this..

**Comment by durack1** [comment]
@doutriaux1 thanks for this!

File: CMake/cdat_modules/pyspharm_pkg.cmake (hunk 1 above)
Comment (sankhesh) [inline CMake/cdat_modules/pyspharm_pkg.cmake]: @doutriaux1 

FYI: #1471

**Pull request merged**